### PR TITLE
RetMgr: Handle cleanup of consumer-groups and consumer-group extents

### PR DESCRIPTION
**Pull request by kirg**

### CODE REVIEW

File: services/retentionmgr/retention.go [hunk 1]
```diff
@@ -312,26 +314,101 @@ func (t *RetentionManager) runRetention(jobsC chan<- *retentionJob) bool {
...
+
+				e := t.metadata.DeleteConsumerGroupExtent(dest.id, cg.id, extID)
+
+				if e != nil {
+					// 'EntityNotExistsError' indicates the CG-extent was already deleted
+					if _, ok := e.(*shared.EntityNotExistsError); ok {
+						e = nil
+					}
+
+					t.logger.WithFields(bark.Fields{
```
Comment (datoug) [inline services/retentionmgr/retention.go]: You can skip the log if err is nil

File: services/retentionmgr/retention.go [hunk 2]
```diff
@@ -312,26 +314,101 @@ func (t *RetentionManager) runRetention(jobsC chan<- *retentionJob) bool {
 			continue
 		}
 
+		var numCGs int
+
+		// query consumer groups for the destination
+		cgs := t.metadata.GetConsumerGroups(dest.id)
+
+		// for each extent, compute retention cursor and convey to storage
```
Comment (datoug) [inline services/retentionmgr/retention.go]: update comment

File: services/retentionmgr/metadataDep.go [hunk 3]
```diff
@@ -352,6 +394,32 @@ func (t *metadataDepImpl) DeleteConsumerGroupExtent(cgID consumerGroupID, extID
 	return nil
 }
 
+func (t *metadataDepImpl) DeleteConsumerGroup(destID destinationID, cgID consumerGroupID) error {
```
Comment (datoug) [inline services/retentionmgr/metadataDep.go]: Suggest to keep the name as the same as metadata. DeleteConsumerGroup in metadata is a different API

File: services/retentionmgr/metadataDep.go [hunk 4]
```diff
@@ -324,7 +324,48 @@ func (t *metadataDepImpl) MarkExtentConsumed(destID destinationID, extID extentI
...
+	})
+
+	for {
+		log.Debug("GetExtentsForConsumerGroup: ReadConsumerGroupExtentsLite on metadata")
+
+		res, e := t.metadata.ReadConsumerGroupExtentsLite(ctx, req)
+		if e != nil {
+			err = e
+			log.WithField(common.TagErr, err).Error("GetExtentsForConsumerGroup: ReadConsumerGroupExtentsLite failed")
+			break
```
[the remark this answers is not in the document]
Comment (kirg) [reply services/retentionmgr/metadataDep.go]: i do a 'break' so as to print the "done" log below .. in my other change (#310), i have added error information to the "done" log too ..

File: services/retentionmgr/metadataDep.go (hunk 3 above)
Comment (kirg) [reply services/retentionmgr/metadataDep.go]: done.

File: services/retentionmgr/metadataDep.go [hunk 5]
```diff
@@ -352,6 +394,32 @@ func (t *metadataDepImpl) DeleteConsumerGroupExtent(cgID consumerGroupID, extID
 	return nil
 }
 
+func (t *metadataDepImpl) DeleteConsumerGroup(destID destinationID, cgID consumerGroupID) error {
+
+	req := metadata.NewDeleteConsumerGroupUUIDRequest()
+	req.UUID = common.StringPtr(string(destID))
+
+	ctx, cancel := thrift.NewContext(2 * time.Second)
```
[the remark this answers is not in the document]
Comment (kirg) [reply services/retentionmgr/metadataDep.go]: i wanted to differentiate between "point" queries (that read/update one row) vs "list" queries (that work on multiple rows). i have made two separate constants and updated code.

**Comment by kirg** [comment]
Discarding this PR. #310 includes all of the changes here and supersedes this.

**Pull request closed without merging**